### PR TITLE
Sharding

### PR DESCRIPTION
**Pull request by vidhy**

### CODE REVIEW

File: src/main/java/io/appform/dropwizard/actors/actor/ActorConfig.java [hunk 1]
```diff
@@ -79,4 +80,12 @@
     @Valid
     private TtlConfig ttlConfig;
 
+    @Min(2)
+    @Max(32)
+    private Integer shardCount;
+
+    public boolean isSharded() {
```
[the remark this answers is not in the document]
Comment (vidhy) [reply src/main/java/io/appform/dropwizard/actors/actor/ActorConfig.java:87]: Added validation.

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 2]
```diff
@@ -99,15 +131,22 @@ public void start() throws Exception {
         this.publishChannel = connection.newChannel();
         connection.ensure(queueName + "_SIDELINE", queueName, dlx,
                 connection.rmqOpts(config));
-        connection.ensure(queueName,
-                config.getExchange(),
-                connection.rmqOpts(dlx, config));
+        if (config.isSharded()) {
+            int bound = config.getShardCount();
+            for (int shardId = 0; shardId < bound; shardId++) {
+                connection.ensure(queueName + "_" + shardId, config.getExchange(), connection.rmqOpts(dlx, config));
```
[the remark this answers is not in the document]
Comment (vidhy) [reply src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java]: Done.

File: src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java [hunk 3]
```diff
@@ -174,4 +185,13 @@ public String visit(DefaultConnectionStrategy strategy) {
         });
     }
 
+    private void validateActorConfig(ActorConfig actorConfig) {
```
Comment (r0goyal) [inline src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java:188]: This can be added as a validationMethod in the config class itself

Comment (vidhy) [reply src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java:188]: Do you suggest moving the method to ActorConfig, but having the invocation in UnmanagedBaseActor. (as it is right now.)?

Comment (vidhy) [reply src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java]: done.

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 4]
```diff
@@ -65,7 +65,17 @@ public final void publish(Message message) throws Exception {
     }
 
     public final void publish(Message message, AMQP.BasicProperties properties) throws Exception {
-        publishChannel.basicPublish(config.getExchange(), queueName, properties, mapper().writeValueAsBytes(message));
+        String routingKey;
+        if (config.isSharded()) {
```
Comment (r0goyal) [inline src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java:69]: This is a repeated code at 2 places. Create a util method which can take actor config and name as input and determine final queue name

**Review by r0goyal — APPROVED** [review]

**Pull request merged**